### ISSUE TITLE
[VPAT][A11y] Radio button should have appropriate state and role announcement

### ISSUE THREAD

**Issue by chunhtai** [flutter/flutter#173054]
b/429089622
Run the https://github.com/flutter/flutter/tree/master/dev/a11y_assessments and navigate to radiolisttile demo

<img width="1020" height="1800" alt="Image" src="https://github.com/user-attachments/assets/b395495d-9f2f-4caa-b27b-657974a0ce5c" />

EXPECTED RESULTS:

With VoiceOver on, swipe to a "Lafayette" radio button in a radio button set.

VoiceOver announces: “Lafayette, radiobutton, checked".

We should double check with cloud a11y team to ensure this is indeed the correct expectation. If so, we may need to maintain the localization for these string and append them to the radio's label
